WalletWasabi.Gui/Global.cs

```diff
@@ -133,6 +133,20 @@ public Global(string dataDir, string torLogsFile, Config config, UiConfig uiConf
 		public async Task InitializeNoWalletAsync()
 		{
 			InitializationStarted = true;
+
+			#region ProcessKillSubscription
+
+			AppDomain.CurrentDomain.ProcessExit += (s, e) => DisposeAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
I would add logging before the `DisposeAsync` call and after it. This is not common way to stop the application, so it's nice to know that it finished processing `DisposeAsync` in its entirety.

Also, your comment from the PR text should be here. Somebody may break this code by reverting to original code, as this looks weird.

Reply:
ACK